test/test_common.rb

```diff
@@ -74,4 +74,11 @@ def test_format_datetime_with_time
     formatted = client.send(:format_datetime, value)
     assert_equal(formatted, '2018-04-30T19:20:33.000000000+00:00')
   end
+
+  def test_parse_definition_with_dashes_in_resource_name
+    method_names = Kubeclient::ClientMixin.parse_definition('TestNameWithDash',\
+                                                            'test-name-with-dashes').method_names
+    assert_equal(method_names[0], 'testnamewithdash')
+    assert_equal(method_names[1], 'test_name_with_dashes')
```

Comment:
Wait, I missed this before, so we'll have `get_testnamewithdash` method for single entity but `get_test_name_with_dashes` for the collection?
I don't like this.

- One direction is for this case to want consistent in `get_test_name_with_dash`, `get_test_name_with_dashes`.
  - one way we could obtain this is to use `[singular_underscores, name.tr('-', '_')]`.  IFF kind's word boundaries match the plural dashes, a bit tricky to detect this...
  - another way is to use singularName!  IFF it's specified, AND we're in "something weird" case so we give up relying on `kind`.  `[singular_name.tr('-', '_'), name.tr('-', '_')]`.

- another direction is to drop the dashes, giving us `get_testnamewithdash`, `get_testnamewithdashes`.

The first direction is prettier — after all somebody went to non-standard trouble :wink: :innocent: to give us word boundaries in both singular and plural case — but I'm not sure how to detect this situation reliably.

PS. I think *predictable* method names are more important consideration than *pretty*.  A dumber algorithm may be better if it's easier to explain?

Reply:
@cben, have a look at the updated  version and see if that's make sense to have.
Added few examples to test.
Please let me know if you'd like to include more cases or alter this PR.